lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -62,28 +62,42 @@ def self.canonicalize_url(driver_url, config)
     # Load balancer methods
     def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
       lb_options ||= {}
-      if lb_options[:security_group_id]
-        security_group = ec2.security_groups[:security_group_id]
-      elsif lb_options[:security_group_name]
-        security_group = ec2.security_groups.filter('group-name', lb_options[:security_group_name])
+      if lb_options[:security_group_ids]
+        security_groups = ec2.security_groups.filter('group-id', lb_options[:security_group_ids]).to_a
+      elsif lb_options[:security_group_names]
+        security_groups = ec2.security_groups.filter('group-name', lb_options[:security_group_names]).to_a
+      else
+        security_groups = []
       end
+      security_group_ids = security_groups.map { |sg| sg.id }
 
-      availability_zones = lb_options[:availability_zones]
+      availability_zones = lb_options[:availability_zones] || []
+      subnets = lb_options[:subnets] || []
       listeners = lb_options[:listeners]
+      scheme = lb_options[:scheme]
+
+      if !availability_zones.empty? && !subnets.empty?
+        raise "You cannot specify both `availability_zones` and `subnets`"
+      end
```

Comment:
The driver DOES give a good error message on this.  With an older version of my code I was seeing some weird issues switching from availability zones to subnets, but I think I prevent that error now.  Removing this.